code/drasil-code/lib/Language/Drasil/Choices.hs

```diff
@@ -362,4 +367,27 @@ choicesSent chs = map chsFieldSent [
 
 -- | Helper to combine pairs of 'Sentence's for rendering 'Choices'.
 chsFieldSent :: (Sentence, Sentence) -> Sentence
-chsFieldSent (rec, chc) = rec +:+ S "selected as" +:+. chc
\ No newline at end of file
+chsFieldSent (rec, chc) = rec +:+ S "selected as" +:+. chc
+
+-- | List of user defined function names.
+-- List is populated with default values.
+fnList :: Map InternalConcept Name
+fnList = fromList [
+  (GetInput, "get_input"), 
+  (DerivedValues, "derived_values"), 
+  (InputConstraints, "input_constraints"), 
+  (WriteOutput, "write_output")]
+
+-- | Data type of user defined concepts
+data InternalConcept = InputConstraints | WriteOutput | DerivedValues 
+  | GetInput | InputParameters | InputFormat deriving (Eq, Ord)  
+
+-- | Returns user defined function Name
+genICFuncName :: InternalConcept -> Name
+genICFuncName ic = existsFuncName(Data.Map.lookup ic (functionNames defaultChoices))
```

Comment:
May as well use `(!)` from `Data.Map` here, if you're just going to error out. 

We need a better way to error out, but forcing that on this PR is too much.

Reply:
This is addressed in a986e0cf111cee60ab034fb92462537589e8bdcd